while-loops/while-loops.md

```diff
@@ -0,0 +1,257 @@
+    i += 1
+
+print("We are outside the while loop! Wooo!")
+```
+
+Here, we have a variable `i` . While `i` is less than or equal to 5, we will print `f"The value of i is {i}"` . Then, we will increment `i` by one.
+
+`i` begins with a value of `1` . As the while loop loops, `i` grows bigger and bigger, since we're modifying the same `i` variable.
+
+At some point, `i` will have a value of `6` . Then, `i <= 5` is falsey, and the while loop will exit. Our code execution moves on to the final `print` statement.
+
+Another way to read this while loop is with a table:
+
+| Value of `i` at the beginning of the loop | Is `i <= 10` ? | Value of `i` at the end of the loop body |
+|-------------------------------------------|---------------|------------------------------------------|
+| `1` | `True` | `2` |
+| `2` | `True` | `3` |
+| `3` | `True` | `4` |
+| `4` | `True` | `5` |
+| `5` | `True` | `6` |
+| `6` | `False` | Loop body doesn't execute                |
+
+Here, we should observe that at some point, `i` has a value of `6` , and the code asked "Is 6 <= 5? Is that true?" Because the answer is false, the code skips over the rest of the loop.
+
+Callout: While loops can accidentally run infinitely! If the conditional for the while loop never becomes false, then the while loop and the program keep going, and going, and going... until something forces it to stop (such as a human closing down the program), or our machine cannot run it anymore. We call these cases **infinite loops, ** and are always a bug.
+
+## Used While Problem-Solving
+
+Let's explore different examples that apply while loops.
+
+We can use a while loop even if we are decrementing a variable...
+
+``` python
+print("Preparing for blastoff in")
+
+countdown = 10
+
+while countdown > 0:
+    print(f"{countdown}...")
+    countdown -= 1
+
+print("Launch!")
+```
+
+We can use a while loop that uses a function to calculate something...
+
+``` python
+flower_basket = []
+
+while len(flower_basket) < 12:
+    flower_basket.append("flower")
+
+print("I have a very full flower basket with a dozen flowers:", flower_basket)
+```
+
+Callout: Note that this conditional uses `<` , _not_ `<=` , because we don't want to add another flower if the length of `flower_basket` is _equal_ to twelve. Watch out for an off-by-one error!
+
+We can use a while loop that only checks if a variable is truthy!
+
+``` python
+import random
+coin_is_tails = True
+
+while coin_is_tails:
+    print("The coin flip is tails!")
+    print("Will we get heads in the next coin flip?")
+    coin_is_tails = bool(random.randint(0, 1))
+
+print("We finally got heads.")
+```
+
+### Counter Variables
+
+Let's return to our original example:
+
+``` python
+i = 1
+while i <= 5:
+    print(f"The value of i is {i}")
+    i += 1
+
+print("We are outside the while loop! Wooo!")
+```
+
+A variable whose job it is to increment or decrement, specifically in a loop, is often called a _counter variable._
```

Comment:
Nit: consider rewording to "A variable that is only incremented or decremented, ...". Saying that a variable _does_ something sounds a little strange.